### ISSUE TITLE
equivalence bounds are tested on the link scale under the default GLM type

### ISSUE THREAD

**Issue by vincentarelbundock** [vincentarelbundock/marginaleffects#1760]
For GLMs, `predictions()` with the default `type` (`"invlink(link)"`) computes estimates on the link scale and back-transforms only `estimate`, `conf.low`, and `conf.high`. The `equivalence` test is applied *before* that back-transform (`finalize_estimates()` in `r/R/utils.R` calls `equivalence()` before `backtransform(pre_transform = linv)`), so user-supplied bounds are compared to a logit-scale estimate while the printed estimate is a probability.

```r
g <- glm(am ~ hp + wt, mtcars, family = binomial)
a <- predictions(g, newdata = datagrid(hp = 150), equivalence = c(0.3, 0.5))
b <- predictions(g, newdata = datagrid(hp = 150), type = "response", equivalence = c(0.3, 0.5))
data.frame(a)[, c("estimate", "p.value.noninf", "p.value.nonsup")]
#    estimate p.value.noninf p.value.nonsup
# 1 0.1542144       0.970434     0.01895852
data.frame(b)[, c("estimate", "p.value.noninf", "p.value.nonsup")]
#    estimate p.value.noninf p.value.nonsup
# 1 0.1542144       0.853982     0.00622431
```

Same displayed estimate, same bounds, different TOST p-values. The default-type call tests H0 on `qlogis(0.154)` against bounds 0.3 and 0.5 interpreted in logits. The same ordering applies with an explicit `transform=`, which is at least implied by the order of operations, but the invlink case is invisible to the user.

Possible fixes:
- When `link_to_response` is `TRUE`, map `equivalence` through `insight::link_function(model)` before calling `equivalence()`.
- Or run `equivalence()` after back-transform using a delta-method SE on the response scale.
- Document that bounds are interpreted on the pre-`transform` scale.

Relevant code: `r/R/predictions.R` (lines ~243-254, 350-380), `r/R/utils.R:69-71`, `r/R/equivalence.R`.
